website/project/decorators.py

```diff
@@ -247,6 +247,30 @@ def wrapped(*args, **kwargs):
 must_be_contributor_or_public_but_not_anonymized = _must_be_contributor_factory(include_public=True, include_view_only_anon=False)
 
 
+def must_be_contributor_even_if_public(func):
```

Comment:
There's one other use of the `@must_be_contributor` decorator, in the `project_remove_contributor` view function. Also not something we want A/VOL users to have access to.

It seems like either a) the default for `include_view_only_anon` in the decorator factory should be `False`, and passed in as `True` for `must_be_contributor_or_public` or b) it should be explicitly defined in the three places that factory is used.

Also, additional logic needs to be added to raise a `403 FORBIDDEN` in the correct case (`not (allow_anon & user_is_anon) & not public_node & not user_is_contrib`)